### ISSUE TITLE
[java] UnusedAssignment false-positive in for-loop with continue

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#5504]
**Affects PMD Version:**
7.10.0

**Rule:** 
https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedassignment

**Code Sample demonstrating the issue:**

```java
List<String> lines = Files.readAllLines("someFile");
int lineNumber = 0;
boolean expectingAnotherLine = false;
for (String line : lines)
{
	++lineNumber;
	if (line.endsWith("\\"))
	{
		expectingAnotherLine = true; // line 9 - false positive unused assignment
		continue;
	}
	expectingAnotherLine = false;
}
if (expectingAnotherLine)
	throw new IllegalArgumentException("Unexpected end of line: " + command);

```

**Expected outcome:**

PMD reports a violation at line 9 (`expectingAnotherLine = true;`), but that's wrong. That's a false positive.

**Running PMD through:**
Maven plugin 3.26.0
